stdlib/marshal.mli

```diff
@@ -109,6 +109,9 @@ val to_channel : out_channel -> 'a -> extern_flags list -> unit
    when read back on a 32-bit platform.  The [Mashal.Compat_32] flag
    only matters when marshaling is performed on a 64-bit platform;
    it has no effect if marshaling is performed on a 32-bit platform.
+   
+   It raises [Failure "output_value: not a binary channel"] if channel
+   [chan] is not in binary mode.
```

Comment:
Some of the documentation precedes ocamldoc having `@raise`, but for new changes it's nice to use it, even if other functions aren't (yet) updated:

```suggestion
   @raise Failure if [chan] is not in binary mode.
```